include/drivers/spi.h

```diff
@@ -297,6 +297,60 @@ struct spi_config {
 	SPI_CONFIG_DT(DT_DRV_INST(inst), operation_, delay_)
 #endif
 
+/**
+ * @brief Complete SPI DT information
+ *
+ * @param bus is the SPI bus
+ * @param config is the slave specific configuration
+ */
+struct spi_dt_spec {
+	const struct device *bus;
+	struct spi_config config;
+};
+
+#ifndef __cplusplus
+/**
+ * @brief Structure initializer for spi_dt_spec from devicetree
+ *
+ * This helper macro expands to a static initializer for a <tt>struct
+ * spi_dt_spec</tt> by reading the relevant bus, frequency, slave, and cs
+ * data from the devicetree.
+ *
+ * Important: multiple fields are automatically constructed by this macro
+ * which must be checked before use. @ref spi_is_ready performs the required
+ * @ref device_is_ready checks.
+ *
+ * This macro is not available in C++.
+ *
+ * @param node_id Devicetree node identifier for the SPI device whose
+ *                struct spi_dt_spec to create an initializer for
+ * @param operation_ the desired @p operation field in the struct spi_config
+ * @param delay_ the desired @p delay field in the struct spi_config's
+ *               spi_cs_control, if there is one
+ */
+#define SPI_DT_SPEC_GET(node_id, operation_, delay_)		     \
+	{							     \
+		.bus = DEVICE_DT_GET(DT_BUS(node_id)),		     \
+		.config = SPI_CONFIG_DT(node_id, operation_, delay_) \
+	}
+
+/**
+ * @brief Structure initializer for spi_dt_spec from devicetree instance
+ *
+ * This is equivalent to
+ * <tt>SPI_DT_SPEC_GET(DT_DRV_INST(inst), operation_, delay_)</tt>.
+ *
+ * This macro is not available in C++.
+ *
+ * @param inst Devicetree instance number
+ * @param operation_ the desired @p operation field in the struct spi_config
+ * @param delay_ the desired @p delay field in the struct spi_config's
+ *               spi_cs_control, if there is one
+ */
+#define SPI_DT_SPEC_GET_INST(inst, operation_, delay_) \
```

Comment:
I think this API should follow the same naming convention useg, e.g. by GPIO (`GPIO_DT_SPEC_INST_GET` and not `GPIO_DT_SPEC_GET_INST`)